mm2src/trading_api/src/one_inch_api/types.rs

```diff
@@ -0,0 +1,416 @@
+
+        Ok(params)
+    }
+
+    /// Validate params by 1inch rules (to avoid extra requests)
+    fn validate_params(&self) -> MmResult<(), ApiClientError> {
+        validate_slippage(self.slippage)?;
+        validate_fee(&self.fee)?;
+        validate_complexity_level(&self.complexity_level)?;
+        validate_gas_limit(&self.gas_limit)?;
+        validate_parts(&self.parts)?;
+        validate_main_route_parts(&self.main_route_parts)?;
+        Ok(())
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TokenInfo {
+    pub address: Address,
+    pub symbol: String,
+    pub name: String,
+    pub decimals: u32,
+    pub eip2612: bool,
+    #[serde(rename = "isFoT")]
+    pub is_fot: bool,
+    #[serde(rename = "logoURI")]
+    pub logo_uri: String,
+    pub tags: Vec<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct ProtocolInfo {
+    pub name: String,
+    pub part: f64,
+
+    #[serde(rename = "fromTokenAddress")]
+    pub from_token_address: Address,
+
+    #[serde(rename = "toTokenAddress")]
+    pub to_token_address: Address,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ClassicSwapData {
+    /// dst token amount to receive, in api is a decimal number as string
+    #[serde(rename = "dstAmount")]
+    pub dst_amount: String,
+    #[serde(rename = "srcToken")]
+    pub src_token: Option<TokenInfo>,
+    #[serde(rename = "dstToken")]
+    pub dst_token: Option<TokenInfo>,
+    pub protocols: Option<Vec<Vec<Vec<ProtocolInfo>>>>,
+    pub tx: Option<TxFields>,
+    pub gas: Option<u128>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TxFields {
+    pub from: Address,
+    pub to: Address,
+    pub data: String,
+    /// tx value, in api is a decimal number as string
+    pub value: String,
+    /// gas price, in api is a decimal number as string
+    #[serde(rename = "gasPrice")]
+    pub gas_price: String,
+    /// gas limit, in api is a decimal number
+    pub gas: u128,
+}
+
+impl ApiClient {
+    pub async fn get_classic_swap_quote(
+        &self,
+        chain_id: u64,
+        params: QueryParams<'_>,
+    ) -> MmResult<ClassicSwapData, ApiClientError> {
+        let api_url = UrlBuilder::new(self, chain_id, ApiClient::get_quote_method().to_owned())
+            .with_query_params(params)
+            .build()?;
+        let value = Self::call_api(api_url).await?;
+        serde_json::from_value(value).map_err(|err| ApiClientError::ParseBodyError(err.to_string()).into())
+    }
+
+    pub async fn get_classic_swap_tx(
+        &self,
+        chain_id: u64,
+        params: QueryParams<'_>,
+    ) -> MmResult<ClassicSwapData, ApiClientError> {
+        let api_url = UrlBuilder::new(self, chain_id, ApiClient::get_swap_method().to_owned())
+            .with_query_params(params)
+            .build()?;
+
+        let value = Self::call_api(api_url).await?;
+        serde_json::from_value(value).map_err(|err| ApiClientError::ParseBodyError(err.to_string()).into())
+    }
+}
```

Comment:
I think this fits more in `client.rs`, also (nit) these methods could be generically implemented.